core/src/main/java/org/mskcc/cbio/oncokb/validation/validation.java

```diff
@@ -184,38 +214,189 @@ public static void main(String[] args) throws IOException, GeneralSecurityExcept
+        if (levelOfEvidence == null)
+            return new HashSet<>();
+        return EvidenceUtils.getEvidenceByEvidenceTypesAndLevels(MainUtils.getTreatmentEvidenceTypes(), Collections.singleton(levelOfEvidence));
+    }
+
+    private static void printEvidences(Set<Evidence> evidences, SpreadsheetService service, WorksheetEntry entry) throws IOException, ServiceException {
+        URL feedUrl = entry.getListFeedUrl();
+        if (evidences != null && service != null && entry != null) {
+            for (Evidence evidence : evidences) {
+                ListEntry row = new ListEntry();
+
+                setValue(row, "Level", evidence.getLevelOfEvidence().getLevel());
+                setValue(row, "Gene", evidence.getGene().getHugoSymbol());
+
+                List<String> alterationNames = new ArrayList<>();
+                for (Alteration alteration : evidence.getAlterations()) {
+                    if (StringUtils.isNullOrEmpty(alteration.getName())) {
+                        alterationNames.add(alteration.getAlteration());
+                    } else {
+                        alterationNames.add(alteration.getName());
+                    }
                 }
+                setValue(row, "Variants", MainUtils.listToString(alterationNames, ", "));
 
+                setValue(row, "Disease", getCancerType(evidence.getOncoTreeType()));
+                setValue(row, "Drugs", org.apache.commons.lang3.StringUtils.join(EvidenceUtils.getDrugs(Collections.singleton(evidence)), ", "));
+                Set<String> articles = EvidenceUtils.getPmids(Collections.singleton(evidence));
+                setValue(row, "PMIDs", org.apache.commons.lang3.StringUtils.join(articles, ", "));
+                setValue(row, "NumberOfPMIDs", Integer.toString(articles.size()));
+
+                Set<ArticleAbstract> articleAbstracts = EvidenceUtils.getAbstracts(Collections.singleton(evidence));
+                Set<String> abstractContent = new HashSet<>();
+                for (ArticleAbstract aa : articleAbstracts) {
+                    abstractContent.add(aa.getAbstractContent());
+                }
+                setValue(row, "Abstracts", org.apache.commons.lang3.StringUtils.join(abstractContent, ", "));
+
+                service.insert(feedUrl, row);
+            }
+        }
+    }
+
+    private static void printTumorTypeSummary(Set<Evidence> evidences, SpreadsheetService service, WorksheetEntry entry) throws IOException, ServiceException {
+        URL feedUrl = entry.getListFeedUrl();
+        if (evidences != null && service != null && entry != null) {
+            for (Evidence evidence : evidences) {
+                ListEntry row = new ListEntry();
+                setValue(row, "Gene", evidence.getGene().getHugoSymbol());
+
+                List<String> alterationNames = new ArrayList<>();
+                for (Alteration alteration : evidence.getAlterations()) {
```

Comment:
Put this snippet of getting alterationNames into a function. Since the same code is used in printEvidences()